proto/invocation.proto

```diff
@@ -532,3 +541,75 @@ message GetTrendResponse {
 
   bool has_invocation_stat_percentiles = 4;
 }
+
+enum MetricType {
+  UNKNOWN_METRIC = 0;
+  DURATION_USEC_METRIC = 1;
+  CAS_CACHE_MISSES_METRIC = 2;
+  EXECUTION_QUEUE_TIME_USEC_METRIC = 3;
+}
+
+message StatFilter {
+  MetricType metric = 1;
+  int64 min = 2;
+  int64 max = 3;
+}
+
+// Fetches a heatmap for the requested metric--a heatmap is basically a set of
+// columns, where each column represents a histogram for that metric in a window
+// of time.
+message GetStatHeatmapRequest {
+  context.RequestContext request_context = 1;
+  TrendQuery query = 2;
+  MetricType metric = 3;
+  int64 num_metric_buckets = 4;
+}
+
+message HeatmapColumn {
+  int64 timestamp_usec = 1;
+  repeated int64 value = 2;
+}
+
+message GetStatHeatmapResponse {
+  context.ResponseContext response_context = 1;
+  // This array is one entry longer than the number of timestamp buckets, with
+  // the actual buckets being defined by 2 sequential entries in the list.
+  repeated int64 timestamp_bracket = 2;
+  // This array is one entry longer than the number of buckets, with the actual
+  // buckets being defined by 2 sequential entries in the list.
+  repeated int64 bucket_bracket = 3;
+  // Each entry in the heatmap response is a histogram for a given time bucket.
+  repeated HeatmapColumn column = 4;
+}
+
+// A drilldown request fetches a set of charts.  Each chart shows a comparison
+// between two data sets, as defined by the 'filter' field.  The 'selection'
+// value contains all data that matches the filters, and the 'base' set contains
+// all data that *doesn't* match the filter.  The set of charts to be returned
+// is determined automatically based on the requested filters.
+message GetStatDrilldownRequest {
+  context.RequestContext request_context = 1;
+  TrendQuery query = 2;
+  repeated StatFilter filter = 3;
+  MetricType drilldown_metric = 4;
+}
+
+message DrilldownEntry {
+  string label = 1;
+  int64 base_value = 2;
+  int64 selection_value = 3;
+}
+
+message DrilldownChart {
+  AggType drilldown_dimension = 1;
+  repeated DrilldownEntry entry = 2;
+}
+
+message GetStatDrilldownResponse {
+  context.ResponseContext response_context = 1;
+  // Each chart shows a drilldown based on a different aggregation type.
+  repeated DrilldownChart chart = 2;
+  // Total fields that can be used to render percentages, if that's desirable.
+  int64 total_in_base = 3;
+  int64 total_in_selection = 4;
+}
```

Comment:
It doesn't look like these new protos need much from invocation -- wdyt about making a "stat{s}.proto" and putting all this new heatmap jazz in there?

Reply:
The dependency graph here gets a little fun: I'm depending on the `TrendQuery` proto which seems worth sharing, but it's reasonable to argue that that should move over to `stats` as well and probably wouldn't be too gnarly to refactor with this change.

I'm inclined to just pull over everything that I'm using over to here at first, and then refactor the rest separately?

Some, like `AggType` don't even have an obvious home--I might pull that out with `StatFilter` into yet another proto package called `filter` or something like that?

Reply:
I guess I forgot to say that if I'm going to make `stats.proto` then I think it makes sense to eventually pull all of the trends requests over to there too, just don't want to make this too too gnarly.

Reply:
Here's what I came up with--moved trends protos into `stats.proto`, moved stuff that still needs to be depended on by `invocation.proto` into `stat_filter.proto`, updated a boatload of references (which should all be harmless since the field numbers and types are unchanged).  What do you all think?